src/nvim/eval.c

```diff
@@ -6797,48 +6795,72 @@ char *set_cmdarg(exarg_T *eap, char *oldarg)
     len += 7 + 4;  // " ++bad=" + "keep" or "drop"
   }
   if (eap->mkdir_p != 0) {
-    len += 4;
+    len += 4;  // " ++p"
   }
 
   const size_t newval_len = len + 1;
   char *newval = xmalloc(newval_len);
+  int xlen = 0;
 
   if (eap->force_bin == FORCE_BIN) {
-    snprintf(newval, newval_len, " ++bin");
+    xlen = snprintf(newval, newval_len, " ++bin");
   } else if (eap->force_bin == FORCE_NOBIN) {
-    snprintf(newval, newval_len, " ++nobin");
+    xlen = snprintf(newval, newval_len, " ++nobin");
   } else {
     *newval = NUL;
   }
+  if (xlen < 0) {
+    goto error;
+  }
 
   if (eap->read_edit) {
-    STRCAT(newval, " ++edit");
+    xlen = snprintf(newval + xlen, newval_len - (size_t)xlen, " ++edit");
```

Comment:
This doesn't look right. Shouldn't it be `xlen +=` here?

Reply:
Yes, but we need to solve it with a helper variable.